### PR TITLE
Make inputs cancellable

### PR DESCRIPTION
**Pull request by dpgraham**
* Used in /release endpoint
* Add a list of actionobjects that undoes any past actions (e.g.: keyUp 'f' undoes keyDown 'f'.... pointerUp 0 undoes pointerDown 0)

### CODE REVIEW

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/state/ActiveInputSources.java [hunk 1]
```diff
@@ -14,7 +15,7 @@
  */
 public class ActiveInputSources {
     private Map<String, InputSource> inputSources = new HashMap<>();
```
Comment (mykola-mokhnach) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/state/ActiveInputSources.java]: final?

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/state/ActiveInputSources.java [hunk 2]
```diff
@@ -14,7 +15,7 @@
...
-    private static ActiveInputSources globalActiveInputSources;
+    private static Map<String, ActiveInputSources> activeInputSources = new HashMap<>();
```
Comment (mykola-mokhnach) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/state/ActiveInputSources.java]: should the mapping be weak?

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/state/InputStateTable.java [hunk 3]
```diff
@@ -27,11 +31,12 @@ public boolean hasInputState(String id) {
...
-        if (globalInputStateTable == null) {
-            globalInputStateTable = new InputStateTable();
-        }
-        return globalInputStateTable;
+    public void addActionToCancel(ActionObject actionObject) {
+        cancelList.add(actionObject);
+    }
+
+    public List<ActionObject> getCancelList() {
+        return cancelList;
```
Comment (mykola-mokhnach) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/state/InputStateTable.java]: should we return the list or a copy of it?

**Comment by mykola-mokhnach** [comment]
I don't observe any unit tests for actions cancelling.

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/state/ActiveInputSources.java (hunk 2 above)
Comment (dpgraham) [reply espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/state/ActiveInputSources.java]: Yeah good thinking. Probably when sessions are deleted we should be removing these.

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/state/InputStateTable.java (hunk 3 above)
Comment (dpgraham) [reply espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/state/InputStateTable.java]: Should be fine to use the actual list. This only gets accessed when the `/release` endpoint gets called, which is a single-threaded handler.

**Comment by dpgraham** [comment]
@mykola-mokhnach I added them to existing tests....

E.g.)

```java
@Test
    public void shouldAddButtonToDepressedOnDispatchDown() throws AppiumException {
        DummyW3CActionAdapter dummyW3CActionAdapter = new DummyW3CActionAdapter();
        InputStateTable inputStateTable = new InputStateTable();
        PointerInputState pointerInputState = new PointerInputState();

        ActionObject actionObject = new ActionObject(
                "id", InputSourceType.POINTER, POINTER_MOVE, 0
        );
        actionObject.setButton(1);

        assertFalse(pointerInputState.isPressed(1));
        assertTrue(inputStateTable.getCancelList().isEmpty());
        dispatchPointerDown(dummyW3CActionAdapter, "any", actionObject, pointerInputState,
                inputStateTable,null);
        assertEquals(inputStateTable.getCancelList().size(), 1);
        ActionObject cancelObject = inputStateTable.getCancelList().get(0);
        assertEquals(cancelObject.getButton(), 1);
        assertEquals(cancelObject.getSubType(), POINTER_UP);
        assertTrue(pointerInputState.isPressed(1));
    }
```

**Comment by dpgraham** [comment]
It's not part of the current pull request to actually apply the cancellations. That will be part of my next PR. This PR just adds the cancellation actions to a list and does nothing with them.

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/state/InputStateTable.java (hunk 3 above)
Comment (dpgraham) [reply espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/state/InputStateTable.java]: Okay, yeah I can do that.

**Pull request merged**